packages/network/src/config.ts

```diff
@@ -66,27 +68,203 @@ export interface NetworkConfig {
+      rootUrl: 'https://bscscan.com/'
+    }
+  },
+  [ChainId.BSC_TESTNET]: {
+    chainId: ChainId.BSC_TESTNET,
+    name: 'bsc-testnet',
+    title: 'BSC Testnet',
+    testnet: true,
+    blockExplorer: {
+      name: 'BSCScan (Testnet)',
+      rootUrl: 'https://testnet.bscscan.com/'
+    }
+  },
+  [ChainId.OPTIMISM]: {
+    chainId: ChainId.OPTIMISM,
+    name: 'optimism',
+    title: 'Optimism',
+    blockExplorer: {
+      name: 'Etherscan (Optimism)',
+      rootUrl: 'https://optimistic.etherscan.io/'
+    }
+  },
+  [ChainId.OPTIMISM_TESTNET]: {
+    chainId: ChainId.OPTIMISM_TESTNET,
+    name: 'optimism-testnet',
+    title: 'Optimistic Kovan',
+    testnet: true,
+    blockExplorer: {
+      name: 'Etherscan (Optimism Testnet)',
+      rootUrl: 'https://kovan-optimistic.etherscan.io/'
+    }
+  },
+  [ChainId.ARBITRUM]: {
+    chainId: ChainId.ARBITRUM,
+    name: 'arbitrum',
+    title: 'Arbitrum',
+    blockExplorer: {
+      name: 'Arbiscan',
+      rootUrl: 'https://arbiscan.io/'
+    }
+  },
+  [ChainId.ARBITRUM_TESTNET]: {
+    chainId: ChainId.ARBITRUM_TESTNET,
+    name: 'arbitrum-testnet',
+    title: 'Arbitrum Testnet',
+    testnet: true,
+    blockExplorer: {
+      name: 'Arbiscan (Testnet)',
+      rootUrl: 'https://testnet.arbiscan.io/'
+    }
+  },
+  [ChainId.AVALANCHE]: {
+    chainId: ChainId.AVALANCHE,
+    name: 'avalanche',
+    title: 'Avalanche',
+    blockExplorer: {
+      name: 'Snowtrace',
+      rootUrl: 'https://snowtrace.io/'
+    }
+  },
+  [ChainId.AVALANCHE_TESTNET]: {
+    chainId: ChainId.AVALANCHE_TESTNET,
+    name: 'avalanche-testnet',
+    title: 'Avalanche Testnet',
+    testnet: true,
+    blockExplorer: {
+      name: 'Snowtrace (Testnet)',
+      rootUrl: 'https://testnet.snowtrace.io/'
+    }
+  },
+  [ChainId.FANTOM]: {
+    chainId: ChainId.FANTOM,
+    name: 'fantom',
+    title: 'Fantom',
+    blockExplorer: {
+      name: 'FTMScan',
+      rootUrl: 'https://ftmscan.com/'
+    }
+  },
+  [ChainId.FANTOM_TESTNET]: {
+    chainId: ChainId.FANTOM_TESTNET,
+    name: 'fantom-testnet',
+    title: 'Fantom Testnet',
+    testnet: true,
+    blockExplorer: {
+      name: 'FTMScan (Testnet)',
+      rootUrl: 'https://testnet.ftmscan.com/'
+    }
+  },
+  [ChainId.GNOSIS]: {
+    chainId: ChainId.GNOSIS,
+    name: 'gnosis',
+    title: 'Gnosis Chain',
+    blockExplorer: {
+      name: 'Gnosis Chain Explorer',
+      rootUrl: 'https://blockscout.com/xdai/mainnet/'
+    }
+  }
+}
 
 export type ChainIdLike = NetworkConfig | BigNumberish
 
-export type NetworksBuilder = (vars: { [key: string]: any }) => Networks
+export type NetworksBuilder = (vars: { [key: string]: any }) => NetworkConfig[]
 
 export const mainnetNetworks = createNetworkConfig(
   (vars: { [key: string]: any }) => [
     {
-      title: 'Ethereum',
-      name: 'mainnet',
-      chainId: ChainId.MAINNET,
+      ...networks[ChainId.MAINNET],
       ensAddress: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
```

Comment:
ensAddress is a property that isnt really used and only relates to ethereum and ethereum testnets and is always the same constant address across all. would rather just leave it out of the main config for now until such time we can flesh that feature out. 

as for indexerUrl, relayer, etc props. these are really specific to the individual config of the wallet. The config here specifies default networks to connect to when no other configuration is added. Wallet will currently send in its own set of wallet configs with computed values based on env. Felt like leaving these types of properties off the main networks object as it serves for more of a meta data object for the networks that gets mixed into to make a full wallet config. Since we are missing services for 90% of the networks on the list felt like its fine to just leave them off that object for now